pkg/k8s/apis/cilium.io/v2alpha1/bgp_node_types.go

```diff
@@ -0,0 +1,183 @@
+	BGPInstances []CiliumBGPNodeInstance `json:"bgpInstances"`
+}
+
+// CiliumBGPNodeInstance is a single BGP router instance configuration on the node.
+type CiliumBGPNodeInstance struct {
+	// Name is the name of the BGP instance.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// LocalASN is the ASN of this virtual router.
+	// Supports extended 32bit ASNs.
+	//
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=4294967295
+	LocalASN *int64 `json:"localASN,omitempty"`
+
+	// RouterID is the BGP router ID of this virtual router.
+	// This configuration is derived from CiliumBGPNodeConfigOverride resource.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Format=ipv4
+	RouterID *string `json:"routerID,omitempty"`
+
+	// LocalPort is the port on which the BGP daemon listens for incoming connections.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:default=179
+	LocalPort *int32 `json:"localPort,omitempty"`
+
+	// Peers is a list of neighboring BGP peers for this virtual router
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +listType=map
+	// +listMapKey=name
+	Peers []CiliumBGPNodePeer `json:"peers"`
+}
+
+type CiliumBGPNodePeer struct {
+	// Name is the name of the BGP peer.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// PeerAddress is the IP address of the neighbor.
+	// Supports IPv4 and IPv6 addresses.
+	//
+	// +kubebuilder:validation:Pattern=`((^\s*((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))\s*$)|(^\s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:)))(%.+)?\s*$))`
+	PeerAddress *string `json:"peerAddress,omitempty"`
+
+	// PeerASN is the ASN of the peer BGP router.
+	// Supports extended 32bit ASNs
+	//
+	// +kubebuilder:validation:Required
```

Comment:
If this field is a pointer and includes the `omitempty` tag, include `// +kubebuilder:validation:Optional` instead of `// +kubebuilder:validation:Required`.